src/cdk/testing/testbed/fake-events/type-in-element.ts

```diff
@@ -25,6 +35,15 @@ const incrementalInputTypes = new Set([
 /** Characters whose key code doesn't match their character code. */
 const KEYCODE_MISMATCHES: Record<string, number> = {
   ',': COMMA,
+  '.': PERIOD,
```

Comment:
Note that I listed just the most common conflicting characters that may show up in a test string. Our approach for mapping characters to keycodes using `charCodeAt(0)` is fundamentally flawed for many more edge cases. It has been this way since we introduced test harnesses.

Reply:
Is there a better design we ought to use?